lib/northbound.c

```diff
@@ -141,6 +143,9 @@ static int nb_node_new_cb(const struct lysc_node *snode, void *arg)
 	assert(snode->priv == NULL);
 	((struct lysc_node *)snode)->priv = nb_node;
 
+	if (module && module->ignore_cbs)
+		SET_FLAG(nb_node->flags, F_NB_NODE_IGNORE_CBS);
+
```

Comment:
Why are these nodes even created in the first place if the module supports no callbacks? If they serve no purpose this code should be changed to not create the nb_nodes and all the rest of the state etc that goes along with them.

Reply:
This is for the Mgmtd deamon where we need the nb_nodes to populate config data trees and the corresponding schema. This is useful for yang schema-based validations and for fetching the equivalent of "show running config".

Reply:
Libyang2 has constructed it's own tree for validation and the output of "show running config". In fact the nb_node's hang off a "priv" pointer in the corresponding libyang2 schema node.

Reply:
I would like to revist this later. We should be able to do validation w/o requiring a bunch of dummy nodes.